src/components/calendar/calendar.jsx

```diff
@@ -102,8 +102,14 @@ function Calendar(props) {
     setActiveDate(newDate);
   }
 
+  function handleDateSelection(date) {
+    props.onDateSelected(date);
```

Comment:
We should probably swap this line with the one below, or move it into a separate `useImperativeHandle` so that retrieving the value with a future ref `value` API will receive the new value, not the old one.

Reply:
I think the date passed in here is the new value.

Reply:
It looks like we don't have an imperative handle. This can be refactored when we do implement an imperative handle.

Basically, what @Bloodyaugust was trying to point out:

The state of the component is _not_ up-to-date when we call `props.onDateSelected(date)`. Yes, we did use the new `date` value. But the component's state is not representative of it.

This is evident when we expose the value of the calendar on an imperative handle (maybe we will never do this?):

- A React component is basically a state machine
- On an event handler, all we should ever be doing is setting state: `setSelectedDate(date)`
- As a state transition (`useEffect(..., [state])`), we should be calling external callbacks: `props.onDateSelected(...)`

```jsx
function Calendar(props, ref) {
  const [selectedDate, setSelectedDate] = useState()

  useImperativeHandle(ref, () => {
    selectedDate: selectedDate,
  })

  useEffect(() => {
    props.onSelectedDateChange(selectedDate)
  }, [selectedDate])

  function handleSelectedDateChange(event) {
    setSelectedDate(event.target.value)
  }

  return (
    <SomeJsX />
  )
```

Reply:
Thanks for explaining better than I did, @juanca. 😅